rero/rero-ils
# cypress: test circulation scenario B

* Implements the test for circulation scenario B.
* Adds needed fixtures.

Co-Authored-by: Alicia Zangger <alicia.zangger@rero.ch>

## Why are you opening this PR?
Task 1613 of US1489
https://tree.taiga.io/project/rero21-reroils/us/1489?milestone=269576
https://tree.taiga.io/project/rero21-reroils/task/1613?kanban-status=1224894

This is the e2e testing of circulation scenario B: https://github.com/rero/rero-ils/blob/dev/doc/circulation/scenarios.md

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* rero/rero-ils-ui https://github.com/rero/rero-ils-ui/tree/zaa-US1394-adapt-circulation-ui branch
* dev of ng-core

## How to test?

You need to include ng-core and rero-ils-ui into rero-ils:
`cd ng-core`
`npm ci && npm run pack`

`cd rero-ils-ui`
`npm ci && npm install path/to/ng-core.tgz --no-save`
`npm run pack`

`cd rero-ils`
`poetry run bootstrap -t path/to/rero-ils-ui.tgz`
`poetry run setup`

Then run the server with `FLASK_DEBUG=False poetry run server`, otherwise the test will fail.
Check that `FLASK_DEBUG=False` is not overwritten in invenio.cfg.

To run the tests on command line: `poetry run cypress` (all the tests).
Or with a GUI: `cd tests/e2e/cypress`, then `npm run cypress`. Click on `Run all specs` on top right of the interface.
Try different browsers.

Documentation: https://github.com/rero/developer-resources/blob/master/base/cypress.md

## Code review check list
